base/linalg/eigen.jl

```diff
@@ -27,6 +27,40 @@ function getindex(A::Union{Eigen,GeneralizedEigen}, d::Symbol)
     throw(KeyError(d))
 end
 
+"""
+    sort!(F::Base.LinAlg.Eigen; kw...)
+
+Sort the eigenvectors and eigenvalues of an eigenfactorization `F` in place using the eigenvalues
+for comparison.  See also `sort!(::AbstractVector)`.
+
+# Examples
+```jldoctest
+julia> F = eigfact([4.0 1.0; 0 1.0])
+Base.LinAlg.Eigen{Float64,Float64,Array{Float64,2},Array{Float64,1}}([4.0, 1.0], [1.0 -0.316228; 0.0 0.948683])
+
+julia> F[:values]
+2-element Array{Float64,1}:
+ 4.0
+ 1.0
+
+julia> sort!(F)
+Base.LinAlg.Eigen{Float64,Float64,Array{Float64,2},Array{Float64,1}}([1.0, 4.0], [-0.316228 1.0; 0.948683 0.0])
+
+julia> F[:values]
+2-element Array{Float64,1}:
+ 1.0
+ 4.0
+```
+"""
+function sort!(F::Eigen; kw...)
+    perm = sortperm(F[:values]; kw...)
```

Comment:
Seems like we need a default `by=eigsortby`, if `by` wasn't supplied in `kw`, using the `eigsortby` function from #21598.

Otherwise it will fail with `no method matching isless` for complex eigenvalues.

Reply:
This was actually intentional, but maybe the error message is confusing? I wanted to avoid choosing any particular sorting order for complex eigenvalues, since I generally prefer `by=angle` and others might want `by=real` or `by=imag`.

Reply:
I can also see a good argument for having *some* default, and `eigsortby` is a pretty good candidate.

Reply:
This could also be done by dispatching on the `eltype` of the eigenvalues, defaulting to `reim`, without the need to define `eigsortby`.

Reply:
Definitely the caller should be able to supply their own `by`, but I think there should be a default, and `reim` seems like a good a choice for complex values (since it will match the sorting of real values in the case of complex numbers that are purely real).